src/TransientCleaner.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * Helper to easily delete WordPress transient on content save based on user-input configuration.
+ *
+ * @package    studiometa/wp-toolkit
+ * @author     Lucas Simeon <lucas.s@studiometa.fr>
+ * @copyright  2020 Studio Meta
+ * @license    https://opensource.org/licenses/MIT
+ * @since      1.0.0
+ * @version    1.0.0
+ */
+
+namespace Studiometa\WPToolkit;
+
+/**
+ * Clean transient based on configuration.
+ */
+class TransientCleaner {
+	const PREFIX                   = 'wp_transient_cleaner_';
+	const OPTION_STORED_TRANSIENTS = self::PREFIX . 'stored_transients';
+
+	/**
+	 * Transients configuration.
+	 *
+	 * @var array
+	 */
+	private $config = array();
+
+	/**
+	 * Stored transients.
+	 *
+	 * @var array|bool
+	 */
+	private $stored_transients;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
```

Comment:
Could we pass the config in the constructor? It might be easier to use.

```php
$transient_cleaner = new TransientCleaner( array() );
$transient_cleaner->init();
```

Reply:
Having a specific method `set_config` instead of passing it to constructor allow us to instanciate configless class to access others class properties. For example, I just want to access to `stored_transients` to check if there is app specific stored transients in database.

Reply:
Ok